src/hostapi/pulseaudio/pa_linux_pulseaudio_cb.c

```diff
@@ -426,6 +434,13 @@ static int _PaPulseAudio_ProcessAudio(PaPulseAudio_Stream *stream,
 
         PaUtil_EndCpuLoadMeasurement( &stream->cpuLoadMeasurer,
                                       hostFrameCount );
+
+        if( ret )
+        {
+            stream->isActive = 0;
+            pulseaudioLength = length - pulseaudioOutputWritten;
+            goto outputzero;
```

Comment:
I find the use of the goto to jump from inside an infinite loop to the inside of previous conditional be be a bit confusing.

Could you put the code on lines 299 to 319 into a subroutine and then call it in two places.
I think that would be easier to read.

Reply:
You are absolutely correct I'll fix that. I'm sometime too fancy with goto that I use it without noting readability.